lib/signs_ui_web/channels/signs_channel.ex

```diff
@@ -77,4 +71,20 @@ defmodule SignsUiWeb.SignsChannel do
     :ok = push(socket, "auth_expired", %{})
     socket
   end
+
+  @spec with_admin_access(Phoenix.Socket.t(), fun()) :: {:noreply, Phoenix.Socket.t()}
```

Comment:
I think the return type should be:

```elixir
{:noreply, Phoenix.Socket.t()} | {:stop, :normal, Phoenix.Socket.t()}
```

Unfortunately it seems like there is no handy type alias in `Phoenix.Channel` for the type of things that should be returned from `handle_in`.